ftplugin/perl/perl.xpt.vim

```diff
@@ -43,109 +44,261 @@ XPT fornn hidden=1
 
 XPT whilenn hidden=1
 
+" Not perl's way of doing things
+XPT while1 hidden=1
+XPT while0 hidden=1
+
 
 XPT perl " #!/usr/bin/env perl
 #!/usr/bin/env perl
 
-..XPT
+use strict;
+use warnings;
 
 
-XPT xif " .. if ..;
-`expr^ if `cond^;
+..XPT
 
+XPT iif " .. if ..;
+if`$SPcmd^(`$SParg^`condition^`$SParg^)`$BRif^{{
+	`cursor^
+}}
 
-XPT xwhile " .. while ..;
-`expr^ while `cond^;
+XPT wwhen " .. when ..;
+when`$SPcmd^(`$SParg^m/`regex^/`$SParg^)`$BRif^{{
+	`cursor^
+}}
```

Comment:
Why are they `iif` and `wwhen`, but not just `if` and `when`?

Reply:
We already have `when` on its own, please check down. This one is really different(look that the braces), it allows `redo`, `last` and `next` to have effect.

Reply:
we cannot use `if` cause it is already taken and `iif` looks a little bit awkwards so maybe `iff` to ease completion?